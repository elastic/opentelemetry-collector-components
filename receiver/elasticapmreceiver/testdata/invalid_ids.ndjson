{"metadata": {"service": {"name": "1234_service-12a3","node": {"configured_name": "node-123"},"version": "5.1.3","environment": "staging","language": {"name": "ecmascript","version": "8"},"runtime": {"name": "node","version": "8.0.0"},"framework": {"name": "Express","version": "1.2.3"},"agent": {"name": "elastic-node", "activation_method": "some_activation_method","version": "3.14.0"}},"user": {"id": "123user", "username": "bar", "email": "bar@user.com"}, "labels": {"tag0": null, "tag1": "one", "tag2": 2}, "process": {"pid": 1234,"ppid": 6789,"title": "node","argv": ["node","server.js"]},"system": {"hostname": "prod1.example.com","architecture": "x64","platform": "darwin", "container": {"id": "container-id"}, "kubernetes": {"namespace": "namespace1", "pod": {"uid": "pod-uid", "name": "pod-name"}, "node": {"name": "node-name"}}},"cloud":{"account":{"id":"account_id","name":"account_name"},"availability_zone":"cloud_availability_zone","instance":{"id":"instance_id","name":"instance_name"},"machine":{"type":"machine_type"},"project":{"id":"project_id","name":"project_name"},"provider":"cloud_provider","region":"cloud_region","service":{"name":"lambda"}}}}
{"transaction": { "id": "foobarInvalid", "trace_id": "0123456789abcdef0123456789abcdefTooLoooong", "parent_id": "foobarInvalid2", "type": "request", "duration": 32.592981, "timestamp": 1496170407154000, "span_count": { "started": 43 } } }
{"span": {"trace_id": "0123456789abcdef0123456789abcdefTooLoooong", "parent_id": "invalid", "id": "invalid", "child_ids": ["51234abcdef56789"], "transaction_id": "01af25874dec69dd", "name": "GET /api/types", "type": "db.postgresql.query.custom","start": null, "duration": 141.581, "timestamp": 1532976822281000, "outcome": "success"}}