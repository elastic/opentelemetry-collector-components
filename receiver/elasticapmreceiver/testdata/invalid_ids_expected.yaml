resourceSpans:
  - resource:
      attributes:
        - key: service.name
          value:
            stringValue: 1234_service-12a3
        - key: service.version
          value:
            stringValue: 5.1.3
        - key: telemetry.sdk.language
          value:
            stringValue: ecmascript
        - key: telemetry.sdk.name
          value:
            stringValue: ElasticAPM
        - key: deployment.environment.name
          value:
            stringValue: staging
        - key: agent.name
          value:
            stringValue: elastic-node
        - key: agent.version
          value:
            stringValue: 3.14.0
        - key: service.language.name
          value:
            stringValue: ecmascript
        - key: service.language.version
          value:
            stringValue: "8"
    scopeSpans:
      - scope: {}
        spans:
          - attributes:
              - key: timestamp.us
                value:
                  intValue: "1496170407154000"
              - key: event.outcome
                value:
                  stringValue: unknown
              - key: processor.event
                value:
                  stringValue: transaction
              - key: transaction.id
                value:
                  stringValue: foobarInvalid
              - key: transaction.name
                value:
                  stringValue: ""
              - key: transaction.sampled
                value:
                  boolValue: true
              - key: transaction.root
                value:
                  boolValue: false
              - key: transaction.type
                value:
                  stringValue: request
              - key: transaction.result
                value:
                  stringValue: ""
              - key: transaction.duration.us
                value:
                  intValue: "32592"
            endTimeUnixNano: "1496170407186592981"
            parentSpanId: ""
            spanId: ""
            startTimeUnixNano: "1496170407154000000"
            status: {}
            traceId: ""
  - resource:
      attributes:
        - key: service.name
          value:
            stringValue: 1234_service-12a3
        - key: service.version
          value:
            stringValue: 5.1.3
        - key: telemetry.sdk.language
          value:
            stringValue: ecmascript
        - key: telemetry.sdk.name
          value:
            stringValue: ElasticAPM
        - key: deployment.environment.name
          value:
            stringValue: staging
        - key: agent.name
          value:
            stringValue: elastic-node
        - key: agent.version
          value:
            stringValue: 3.14.0
        - key: service.language.name
          value:
            stringValue: ecmascript
        - key: service.language.version
          value:
            stringValue: "8"
    scopeSpans:
      - scope: {}
        spans:
          - attributes:
              - key: timestamp.us
                value:
                  intValue: "1532976822281000"
              - key: event.outcome
                value:
                  stringValue: success
              - key: processor.event
                value:
                  stringValue: span
              - key: span.duration.us
                value:
                  intValue: "141581"
              - key: span.id
                value:
                  stringValue: invalid
              - key: span.name
                value:
                  stringValue: GET /api/types
              - key: span.type
                value:
                  stringValue: db
              - key: span.subtype
                value:
                  stringValue: postgresql
              - key: span.action
                value:
                  stringValue: query.custom
            endTimeUnixNano: "1532976822422581000"
            name: GET /api/types
            parentSpanId: ""
            spanId: ""
            startTimeUnixNano: "1532976822281000000"
            status:
              code: 1
            traceId: ""
